### ISSUE TITLE
MacOS Big Sur - Lens app causes extremely high host CPU usage when power mgmt kicks in and screen turns off.

### ISSUE THREAD

**Issue by mcooknu** [lensapp/lens#1952]
If Lens is open, minimised, or the "Lens Helper" is left running on the MacOS Big Sur laptop, once the screensaver kicks in and the screen turns off the laptop CPU starts to go up to a point that you hear all fans on the laptop at maximum forever until you return to the laptop and screen turns back on.  (laptop screen + monitor configuration).

**To Reproduce**
1. Install Lens
2. Open Lens 
3. Leave laptop alone and wait for screen to turn off as part of power management
4. Wait ~2mins and CPU will cause fans on laptop to go to maximum

**Expected behavior**
Lens does not generate any CPU load and certainly not max out the CPU to the point of overheating the laptop.

**Environment (please complete the following information):**
- Lens Version:
- OS: MacOS Big Sur
- Download from website

If I manually kill all "Lens" related processes, including the Lens Helper I see in the process list, then the laptop does not exhibit this maxed out CPU and fan issue.

**Comment by cathy-kim** [comment]
It also happens on macOS Catalina 😢

**Comment by jim-docker** [comment]
We're having trouble reproducing this, is there any more information you can provide. What cluster(s) are connected? Have you tried looking at the CPU usage with the macOS resource monitor? Is the behaviour consistent or intermittent? What are your power management settings?

**Comment by mcooknu** [comment]
<img width="654" alt="Screen Shot 2021-01-25 at 8 50 35 AM" src="https://user-images.githubusercontent.com/61422545/105714478-654f8300-5eea-11eb-88f2-29617b9ef845.png">

1. Dual-montior (laptop + monitor) setup.
2. Noticed when this happens, that "com.docker.hyperkit" is at 50% CPU when I come back to my laptop and the screen turns on.  I dont see a Lens process with high CPU.
3. I usually always start Lens via command line, using "open -a Lens" after having issues AWS SSO commands to log into the account.
4. To switch between AWS Accounts, I kill Lens, re-issue new SSO for different AWS account then launch it again.
ps -ax  | grep -e "[L]ens" | awk '{print $1}' | xargs -t kill

Typically I only connect to one AWS EKS Cluster... although may move between 2-3 in the same AWS Account.

The behaviour is ~80-90% reproducable and I am now putting my laptop into Sleep mode to avoid the chance of returning to a Mac that sounds like a jet engine.  I really like Lens application so hopefully we can figure this out.

**Comment by jim-docker** [comment]
You mention docker at 50% cpu, I assume you don't need it when accessing AWS clusters from Lens. Have you tried quitting docker desktop just to see if it is related to this problem? Maybe memory is getting exhausted? How big is the swap used?

**Comment by jakolehm** [comment]
@mcooknu does this happen also with Lens v4.1.0?

**Comment by Nokel81** [comment]
Might be related to #1725

**Comment by tomharrisonjr** [comment]
Possible clue?

I have also observed high CPU and am still on 4.2.3 -- we use AWS SSO and in several cases, after coming in in the morning after computer slept, the UI (running on a separate macOS space) displayed what was visible on the previous day for several seconds after switching to that space.  Then the UI refreshed and indicated the usual error `The SSO session associated with this profile has expired or is otherwise invalid.` and the CPU settled shortly thereafter.

I noticed because the fan was running, and Activity Monitor confirmed Lens Renderer was consuming lots of CPU.  macOS Big Sur 11.2.3

**Comment by Nokel81** [comment]
@tomharrisonjr Yes, I think that is what was also realized in #2119 so that is what we will be looking at. Might be related to retrying a bunch of watches really fast

**Comment by Nokel81** [comment]
@tomharrisonjr so if you refresh your SSO, does the problem go away?

**Comment by Nokel81** [comment]
Because if so, then this would be a duplicate of #2119
